oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentials.java

```diff
@@ -53,22 +54,44 @@
 import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Map;
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
 
 /**
- * Url-sourced and file-sourced external account credentials.
+ * Url-sourced, file-sourced, or user provided supplier method-sourced external account credentials.
  *
  * <p>By default, attempts to exchange the external credential for a GCP access token.
  */
 public class IdentityPoolCredentials extends ExternalAccountCredentials {
 
+  static final String FILE_METRICS_HEADER_VALUE = "file";
+  static final String URL_METRICS_HEADER_VALUE = "url";
   private static final long serialVersionUID = 2471046175477275881L;
 
-  private final IdentityPoolCredentialSource identityPoolCredentialSource;
+  @Nullable private final IdentityPoolCredentialSource identityPoolCredentialSource;
+
+  @Nullable private final Supplier<String> subjectTokenSupplier;
```

Comment:
no, since its just a different way to retrieve the subject token. The only place we currently do any additional logic on the subject token after retrieving it is the AwsCredentials (because we take the AWS security credentials and turn it into a signed key). All other current use cases are covered by the logic added to identity pool credentials.